### PR TITLE
feat(EMS-863-963-951): Account - Password reset - form and email

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds the beginnings of the "Password reset" journey.

## Changes

- Rename existing constants from `RESET_PASSWORD` to `PASSWORD_RESET` to be consistent with the actual route, as per design.
- Add new fields to exporter/account keystone schema: `passwordResetHash` and `passwordResetExpiry`.
- Create new custom GQL resolver, `sendEmailPasswordResetLink`. This:
  - Gets an exporters account.
  - Generates a reset password hash with 5mins expiry.
  - Sends an email to the exporter via `sendEmails.passwordResetLink`.
- Setup UI page template, routes, controller and validation for "password reset".
- Create new UI queries to call new GQL resolver, `sendEmailPasswordResetLink`.
- Add E2E test coverage for:
  - Password reset page.
  - Password reset page - form submission/validation.
  - Password reset page - form submission when an account does not exist with the provided email.

### CODE REVIEW

File: src/api/custom-resolvers/send-email-password-reset-link.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    // Get the account the email is associated with.
+    const exporter = await getAccountByField(context, FIELD_IDS.ACCOUNT.EMAIL, email);
+
+    if (!exporter) {
+      console.info('Unable to send password reset email - no account found');
+
+      return { success: false };
+    }
+
+    const passwordResetHash = crypto.pbkdf2Sync(email, exporter.salt, ITERATIONS, KEY_LENGTH, DIGEST_ALGORITHM).toString(STRING_TYPE);
```
Comment (ttbarnes) [inline src/api/custom-resolvers/send-email-password-reset-link.ts:41]: @abhi-markan Keen to get any feedback from you on this please - this is where we generate a password reset hash with 5mins expiry and save it in the DB. Uses same approach/encryption as DTFS.

Next PRs will be about handling/consuming the hash/reset link sent via email.

File: src/api/constants/index.ts [hunk 2]
```diff
@@ -14,6 +14,20 @@ export const ANSWERS = {
 
 export const GBP_CURRENCY_CODE = 'GBP';
 
+/**
+ * DATE_5_MINUTES_FROM_NOW
+ * Generate a date that is 5 minutes from now
+ * @returns {Date}
+ */
+const DATE_5_MINUTES_FROM_NOW = () => {
```
Comment (Zainzzkk) [inline src/api/constants/index.ts:22]: should constants have functions? or should it be in a helper with referral to the milliseconds constant?

Comment (ttbarnes) [reply src/api/constants/index.ts:22]: I don't see why constants shouldn't have functions? We have functions for lots of account things - session expiry, verification expiry, password reset

File: src/api/emails/index.ts [hunk 3]
```diff
@@ -89,6 +89,32 @@ const securityCodeEmail = async (emailAddress: string, firstName: string, securi
...
+
+    const variables = { firstName, passwordResetToken: passwordResetHash };
+
+    const response = await callNotify(templateId, emailAddress, variables);
+
+    return response;
+  } catch (err) {
+    console.error(err);
+
+    throw new Error(`Sending email for account password reset ${err}`);
```
Comment (Zainzzkk) [inline src/api/emails/index.ts:114]: this file is getting quite big.  tech debt to split this into folders and have one index file with imports? Like custom-resolvers?

File: src/ui/server/controllers/insurance/account/password-reset/index.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  },
+} = ROUTES;
+
+/**
+ * PAGE_VARIABLES
+ * Page fields
+ * @returns {Object} Page variables
+ */
+export const PAGE_VARIABLES = {
+  FIELD: {
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/account/password-reset/index.ts:28]: This follows other patterns, the alternative is to have a `FIELD` object or have it directly in the controller definition - I think `PAGE_VARIABLES` is cleaner and it allows for extension in the future. What do you think?

**Review by Zainzzkk — APPROVED** [review]

File: src/api/custom-resolvers/send-email-password-reset-link.ts (hunk 1 above)
Comment (ttbarnes) [reply src/api/custom-resolvers/send-email-password-reset-link.ts]: In the interests of progressing the sprint and unblocking PRs, going to walk through password reset as a whole in one session and can make changes after this if required

**Pull request merged**